packages/builder/builder-rspack-provider/src/plugins/define.ts

```diff
@@ -1,29 +1,32 @@
-import _ from '@modern-js/utils/lodash';
-import type { BuilderPlugin, SourceConfig } from '../types';
-import { setConfig } from '@modern-js/builder-shared';
+import { mapValues } from '@modern-js/utils/lodash';
+import { setConfig, type GlobalVars } from '@modern-js/builder-shared';
+import type { BuilderPlugin } from '../types';
 
 export const builderPluginDefine = (): BuilderPlugin => ({
   name: 'builder-plugin-define',
 
   async setup(api) {
-    api.modifyRspackConfig(async rspackConfig => {
-      const { getNodeEnv, removeTailSlash } = await import('@modern-js/utils');
+    api.modifyRspackConfig(async (rspackConfig, { env, target }) => {
+      const { getNodeEnv, removeTailSlash, applyOptionsChain } = await import(
+        '@modern-js/utils'
+      );
       const config = api.getNormalizedConfig();
 
-      const builtinVars: NonNullable<SourceConfig['globalVars']> = {
+      const builtinVars: GlobalVars = {
         'process.env.NODE_ENV': getNodeEnv(),
         'process.env.ASSET_PREFIX': removeTailSlash(
           rspackConfig.output?.publicPath || '/',
         ),
       };
 
       // Serialize global vars. User can customize value of `builtinVars`.
-      const globalVars = {
-        ...builtinVars,
-        ...(config.source.globalVars ?? {}),
-      };
+      const globalVars = applyOptionsChain(
+        builtinVars,
+        config.source.globalVars,
+        { env, target },
+      );
 
-      const serializedVars = _.mapValues(globalVars, value =>
+      const serializedVars = mapValues(globalVars, value =>
         JSON.stringify(value),
```

Comment:
Rspack 这边好像缺了 undefined 的处理

```suggestion
        JSON.stringify(value) ?? 'undefined',
```

Reply:
确实（两套代码的弊端